tests/Meziantou.Analyzer.Test/Rules/DoNotUseElseWhenIfBlockJumpsAnalyzerTests.cs

```diff
@@ -0,0 +1,136 @@
+﻿using System.Threading.Tasks;
+using Meziantou.Analyzer.Rules;
+using TestHelper;
+using Xunit;
+
+namespace Meziantou.Analyzer.Test.Rules
+{
+    public sealed class DoNotUseElseWhenIfBlockJumpsAnalyzerTests
+    {
+        private const string IfThatContinuesElse = @"
+class TestClass
+{
+    int Test(int value)
+    {
+        while (true)
+        {
+            if (value < 0)
+            {
+                value++;
+                continue;
+            }
+        [||]else
+            {
+                return 1;
+            }
+        }
+    }
+}";
+        private const string IfThatReturnsElse = @"
+class TestClass
+{
+    int Test(int value)
+    {
+        if (value < 0)
+        {
+            // Indicates it's a negative number
+            return -1;
+        }
+    [||]else
+        {
+            // Indicates it's a positive number
+            return 1;
+        }
+    }
+}";
+        private const string IfThatThrowsElse = @"
+class TestClass
+{
+    int Test(int value)
+    {
+        if (value < 0)
+        {
+            throw new System.ArgumentNullException(nameof(value));
+        }
+    [||]else
```

Comment:
I think this is better to detect the reported text. This would ensure the diagnostic doesn't produce large squiggles in the editor.

```suggestion
    [|else|]
```

Reply:
After making this change, my tests were failing. Reverted.

Reply:
`context.ReportDiagnostic(s_rule, elseClauseSyntax);` should select the whole else block. I this you should change it to specify the text span to report.

Reply:
Same comment as below. Resolving this one.